### PR TITLE
SFDL-2451 - SFTP getHomeDir error fixes

### PR DESCRIPTION
**Pull request by pgaliana**
A new getHomeDir method was added. The method returns an empty string when the home directory is "/".
The new method is used in the following methods: getFile, getFileStream, listFiles and putFile.

### CODE REVIEW

File: src/main/java/org/mule/modules/SftpUtils.java [hunk 1]
```diff
@@ -118,11 +122,14 @@ public static void putFile(Session session, ChannelSftp command, InputStream con
...
-            return result;
+            return command.get(getHomeDir(command) + filePath);
         } catch (SftpException e) {
             releaseConnection(session, command);
             throw new SftpLiteException("Error retrieving file stream from SFTP. Original Cause("+e.getMessage()+")");
         }
     }
+
+    private static String getHomeDir(ChannelSftp command) throws SftpException {
+        return command.getHome().equals("/") ? "" : command.getHome();
```
Comment (marcosnc) [inline src/main/java/org/mule/modules/SftpUtils.java]: What happens if the home directory starts with a slash but it also contains a path? Or if it ends with a slash?

Comment (pgaliana) [reply src/main/java/org/mule/modules/SftpUtils.java]: 1. If the home directory starts with "/" but also contains a path (e.g. /some_dir) then we get "/some_dir" and everything is ok.
2. command.getHome never returns a path with a trailing slash, at least in all the SFTP versions we've checked. But, nevertheless, we can check for the trailing slash and remove it.

Comment (marcosnc) [reply src/main/java/org/mule/modules/SftpUtils.java]: Do you want to add the trailing slash check to this PR?

Comment (pgaliana) [reply src/main/java/org/mule/modules/SftpUtils.java]: I added the check.

**Pull request merged**